CMakeLists.txt

```diff
@@ -1,60 +1,217 @@
+##     find_package(catkin REQUIRED COMPONENTS ...)
+##   * add "message_runtime" and every package in MSG_DEP_SET to
+##     catkin_package(CATKIN_DEPENDS ...)
+##   * uncomment the add_*_files sections below as needed
+##     and list every .msg/.srv/.action file to be processed
+##   * uncomment the generate_messages entry below
+##   * add every package in MSG_DEP_SET to generate_messages(DEPENDENCIES ...)
+
+## Generate messages in the 'msg' folder
+# add_message_files(
+#   FILES
+#   Message1.msg
+#   Message2.msg
+# )
+
+## Generate services in the 'srv' folder
+# add_service_files(
+#   FILES
+#   Service1.srv
+#   Service2.srv
+# )
 
-set(HEADER_FILES
-  "include/ahrs/AHRS.h"
-  "include/ahrs/AHRSProtocol.h"
-  "include/ahrs/ContinuousAngleTracker.h"
-  "include/ahrs/IBoardCapabilities.h"
-  "include/ahrs/IIOCompleteNotification.h"
-  "include/ahrs/IIOProvider.h"
-  "include/ahrs/IMUProtocol.h"
-  "include/ahrs/IMURegisters.h"
-  "include/ahrs/InertialDataIntegrator.h"
-  "include/ahrs/ITimestampedDataSubscriber.h"
-  "include/ahrs/OffsetTracker.h"
-  "include/ahrs/SerialIO.h"
-  "include/ahrs/SerialPort.h")
-
-add_library(ahrs SHARED ${SOURCES})
-set_target_properties(ahrs PROPERTIES 
-  PUBLIC_HEADER "${HEADER_FILES}"
-  FRAMEWORK TRUE
+## Generate actions in the 'action' folder
+# add_action_files(
+#   FILES
+#   Action1.action
+#   Action2.action
+# )
+
+## Generate added messages and services with any dependencies listed here
+# generate_messages(
+#   DEPENDENCIES
+#   sensor_msgs
+# )
+
+################################################
+## Declare ROS dynamic reconfigure parameters ##
+################################################
+
+## To declare and build dynamic reconfigure parameters within this
+## package, follow these steps:
+## * In the file package.xml:
+##   * add a build_depend and a exec_depend tag for "dynamic_reconfigure"
+## * In this file (CMakeLists.txt):
+##   * add "dynamic_reconfigure" to
+##     find_package(catkin REQUIRED COMPONENTS ...)
+##   * uncomment the "generate_dynamic_reconfigure_options" section below
+##     and list every .cfg file to be processed
+
+## Generate dynamic reconfigure parameters in the 'cfg' folder
+# generate_dynamic_reconfigure_options(
+#   cfg/DynReconf1.cfg
+#   cfg/DynReconf2.cfg
+# )
+
+###################################
+## catkin specific configuration ##
+###################################
+## The catkin_package macro generates cmake config files for your package
+## Declare things to be passed to dependent projects
+## INCLUDE_DIRS: uncomment this if your package contains header files
+## LIBRARIES: libraries you create in this project that dependent projects also need
+## CATKIN_DEPENDS: catkin_packages dependent projects also need
+## DEPENDS: system dependencies of this project that dependent projects also need
+catkin_package(
+  INCLUDE_DIRS include
+#  LIBRARIES navx
+  CATKIN_DEPENDS sensor_msgs
+#  DEPENDS system_lib
```

Comment:
Threads

Reply:
Catkin seems to be angry about this because Threads doesn't define any standard include dirs: 
```CMake Warning at /opt/ros/melodic/share/catkin/cmake/catkin_package.cmake:166 (message):
  catkin_package() DEPENDS on 'Threads' but neither 'Threads_INCLUDE_DIRS'
  nor 'Threads_LIBRARIES' is defined.
Call Stack (most recent call first):
  /opt/ros/melodic/share/catkin/cmake/catkin_package.cmake:102 (_catkin_package)
  navx/CMakeLists.txt:31 (catkin_package)
```

Is this safe to ignore?

Reply:
Actually, instead of having multiple CATKIN_DEPENDS, you just have one with all the packages after it (doesn't have to be on same line, but it reads until the next separator (DEPENDS or end parenth) (this might've been the bug).
If you comment it out, that should be fine I think, as long as it links against it in the compiling. Easy test.